orttraining/orttraining/core/graph/optimizer/lamb_optimizer_builder.cc

```diff
@@ -75,7 +76,7 @@ Status LambOptimizerBuilder::Build(
   } else {
     step_tensor_proto = CreateTensorProto<int64_t>(step_tensor_name, 1);
   }
-  new_external_initializers.emplace_back(step_tensor_proto);
+  weight_to_opt_mapping[onnxruntime::training::SHARED_OPTIMIZER_STATES_KEY] = {step_tensor_proto};
```

Comment:
Does ```TensorProto``` have the name of the tensor embedded inside it? For example, for shared optimizers, the structure is:
```
{
    "shared_optimizer_state":
    {
        "step": step_array
    }
}
```

So, either the second level is missing here or the ```TensorProto``` has the 'step' name embedded within it?

Reply:
Yes, this is exactly where `["shared_optimizer_state"]["step"]` is added to `weight_to_opt_mapping`